examples/tv-casting-app/linux/main.cpp

```diff
@@ -0,0 +1,140 @@
+    InitServer();
+    ReturnOnFailure(OpenBasicCommissioningWindow(ResetFabrics::kYes, 3 * 60));
+
+    // Display onboarding payload
+    chip::DeviceLayer::ConfigurationMgr().LogDeviceConfig();
+
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    if (selectedCommissioner != nullptr)
+    {
+        // Advertise self as Commissionable Node over mDNS
+        ReturnOnFailure(app::Mdns::AdvertiseCommissionableNode(app::Mdns::CommissioningMode::kEnabledBasic));
+
+        // Send User Directed commissioning request
+        ReturnOnFailure(SendUserDirectedCommissioningRequest(chip::Transport::PeerAddress::UDP(
+            selectedCommissioner->ipAddress[0], selectedCommissioner->port, selectedCommissioner->interfaceId[0])));
+    }
+#endif // CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+}
+
+/**
+ * Accepts user input of selected commissioner and calls PrepareForCommissioning with
+ * the selected commissioner
+ */
+void RequestUserDirectedCommisisioning(System::SocketEvents events, intptr_t data)
+{
+    // Accept user selection for commissioner to request commissioning from
+    int selectedCommissionerNumber = CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES;
+    scanf("%d", &selectedCommissionerNumber);
+    chip::DeviceLayer::SystemLayer.StopWatchingSocket(&token);
+
+    const Mdns::DiscoveredNodeData * selectedCommissioner =
+        commissionableNodeController.GetDiscoveredCommissioner(selectedCommissionerNumber - 1);
+    VerifyOrReturn(selectedCommissioner != nullptr, ChipLogError(Zcl, "No such commissioner!"));
+    PrepareForCommissioning(selectedCommissioner);
+}
+
+void InitCommissioningFlow(intptr_t commandArg)
+{
+    int commissionerCount = 0;
+
+    // Display discovered commissioner TVs to ask user to select one
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        const Mdns::DiscoveredNodeData * commissioner = commissionableNodeController.GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr)
+        {
+            ChipLogProgress(Zcl, "Discovered Commissioner #%d", ++commissionerCount);
+            commissioner->LogDetail();
+        }
+    }
+
+    if (commissionerCount > 0)
+    {
+        ChipLogProgress(
+            Zcl, "%d commissioner(s) discovered. Select one (by number# above) to request commissioning from: ", commissionerCount);
+
+        // Setup for async/non-blocking user input from stdin
+        int flags = fcntl(0, F_GETFL, 0);
+        VerifyOrReturn(fcntl(0, F_SETFL, flags | O_NONBLOCK) == 0,
+                       ChipLogError(Zcl, "Could not set non-blocking mode for user input!"));
+        ReturnOnFailure(chip::DeviceLayer::SystemLayer.StartWatchingSocket(0, &token));
+        ReturnOnFailure(chip::DeviceLayer::SystemLayer.SetCallback(token, RequestUserDirectedCommisisioning,
+                                                                   reinterpret_cast<intptr_t>(&commissionerCount)));
```

Comment:
The passed data happens not to be used in `RequestUserDirectedCommisisioning`.  Still, it is incorrect to pass a pointer to commissionerCount, which is on the stack.

```suggestion
                                                                   nullptr));
```

Reply:
Thanks for catching this. Actually we don't need any data passed to the callback in this case. So I have updated the diff to just pass in a null.